### PR TITLE
add pathspec 0.5.9

### PR DESCRIPTION
**Pull request by bollwyvl**
Adds [pathspec](https://github.com/cpburnz/python-path-specification) 0.5.9.

This is a missing dependency for https://github.com/conda-forge/yamllint-feedstock/issues/5

@proinsias interested in co-maintaining?

### CODE REVIEW

**Comment by bollwyvl** [comment]
Not sure what's up with azure/win... been seeing this a few times. Also added the `doc_url` and `doc_source_url`, but will wait to see how other builds shake out.

File: recipes/pathspec/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+test:
+  requires:
+    - pytest
+  imports:
+    - pathspec
+  commands:
+    - pytest --pyargs pathspec  # [unix]
+    # skip symlink tests on windows
+    # https://github.com/cpburnz/python-path-specification/commit/d268375a4c6aa
+    - pytest --pyargs pathspec -k "not test_2_1_check_realpath"  # [win]
```
Comment (bollwyvl) [inline recipes/pathspec/meta.yaml]: Here's the fail from [appveyor](https://ci.appveyor.com/project/conda-forge/staged-recipes/builds/22388359#L872): I guess I'm not too surprised about symlinks misbehaving there. I see no harm in skipping it with `-k not test_2_1_check_realpath`
```
================================== FAILURES ===================================
____________________ IterTreeTest.test_2_1_check_realpath _____________________
self = <pathspec.tests.test_util.IterTreeTest testMethod=test_2_1_check_realpath>
    def test_2_1_check_realpath(self):
    	"""
    	Tests whether `os.path.realpath` works properly with symlinks.
    	"""
    	# NOTE: Windows does not follow symlinks with `os.path.realpath`
    	# which is what we use to detect recursion. See <https://bugs.python.org/issue9949>
    	# for details.
    	broken_realpath = None
    	try:
    		self.require_symlink()
    		file = os.path.join(self.temp_dir, 'file')
    		link = os.path.join(self.temp_dir, 'link')
    		self.mkfile(file)
    		os.symlink(file, link)
    
    		try:
>   			self.assertEqual(os.path.realpath(file), os.path.realpath(link))
E      AssertionError: 'C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\tmpras3doid\\file' != 'C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\tmpras3doid\\link'
E      - C:\Users\appveyor\AppData\Local\Temp\1\tmpras3doid\file
E      ?                                                    ^ ^^
E      + C:\Users\appveyor\AppData\Local\Temp\1\tmpras3doid\link
E      ?                                                    ^ ^^
..\_test_env\lib\site-packages\pathspec\tests\test_util.py:154: AssertionError
=============== 1 failed, 38 passed, 4 skipped in 0.19 seconds ================
(%PREFIX%) %SRC_DIR%>IF 1 NEQ 0 exit /B 1 

```

**Comment by bollwyvl** [comment]
Aha, the azure fail is not a problem:
https://github.com/conda-forge/staged-recipes/issues/7715#issuecomment-460897996

**Comment by bollwyvl** [comment]
@help-python i think we're ready for review here

File: recipes/pathspec/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    pathspec is a utility library for pattern matching of file paths. So far
+    this only includes Git's wildmatch pattern matching which itself is derived
+    from Rsync's wildmatch. Git uses wildmatch for its gitignore files.
+  doc_url: https://python-path-specification.readthedocs.io
+  doc_source_url: https://github.com/cpburnz/python-path-specification/blob/master/doc/source/index.rst
+
+extra:
+  recipe-maintainers:
+    - bollwyvl
+    - proinsias
```
Comment (chrisburr) [inline recipes/pathspec/meta.yaml]: @proinsias Are you happy to be listed as a maintainer?

Comment (bollwyvl) [reply recipes/pathspec/meta.yaml]: thanks for being thorough: also have this downstream:
https://github.com/conda-forge/yamllint-feedstock/pull/12#issuecomment-463885519

Comment (chrisburr) [reply recipes/pathspec/meta.yaml]: Perfect, LGTM 👍

**Comment by bollwyvl** [comment]
Thanks!

**Pull request merged**